dgl_sparse/src/sparse_format.cc

```diff
@@ -14,21 +14,77 @@
 namespace dgl {
 namespace sparse {
 
-std::shared_ptr<COO> CSRToCOO(
-    int64_t num_rows, int64_t num_cols, const std::shared_ptr<CSR> csr) {
-  auto indptr = TorchTensorToDGLArray(csr->indptr);
-  auto indices = TorchTensorToDGLArray(csr->indices);
-  bool data_as_order = false;
-  runtime::NDArray data = aten::NullArray();
-  if (csr->value_indices.has_value()) {
-    data_as_order = true;
-    data = TorchTensorToDGLArray(csr->value_indices.value());
-  }
-  auto dgl_csr = aten::CSRMatrix(num_rows, num_cols, indptr, indices, data);
-  auto dgl_coo = aten::CSRToCOO(dgl_csr, data_as_order);
+std::shared_ptr<COO> COOFromOldDGLCOO(const aten::COOMatrix& dgl_coo) {
   auto row = DGLArrayToTorchTensor(dgl_coo.row);
   auto col = DGLArrayToTorchTensor(dgl_coo.col);
-  return std::make_shared<COO>(COO{row, col});
+  auto data = DGLArrayToOptionalTorchTensor(dgl_coo.data);
+  return std::make_shared<COO>(
+      COO{dgl_coo.num_rows, dgl_coo.num_cols, row, col, data,
+          dgl_coo.row_sorted, dgl_coo.col_sorted});
+}
+
+aten::COOMatrix COOToOldDGLCOO(const std::shared_ptr<COO>& coo) {
+  auto row = TorchTensorToDGLArray(coo->row);
+  auto col = TorchTensorToDGLArray(coo->col);
+  auto data = OptionalTorchTensorToDGLArray(coo->data);
+  return aten::COOMatrix(
+      coo->num_rows, coo->num_cols, row, col, data, coo->row_sorted,
+      coo->col_sorted);
+}
+
+std::shared_ptr<CSR> CSRFromOldDGLCSR(const aten::CSRMatrix& dgl_csr) {
+  auto indptr = DGLArrayToTorchTensor(dgl_csr.indptr);
+  auto indices = DGLArrayToTorchTensor(dgl_csr.indices);
+  auto data = DGLArrayToOptionalTorchTensor(dgl_csr.data);
+  return std::make_shared<CSR>(
+      CSR{dgl_csr.num_rows, dgl_csr.num_cols, indptr, indices, data,
+          dgl_csr.sorted});
+}
+
+aten::CSRMatrix CSRToOldDGLCSR(const std::shared_ptr<CSR>& csr) {
+  auto indptr = TorchTensorToDGLArray(csr->indptr);
+  auto indices = TorchTensorToDGLArray(csr->indices);
+  auto data = OptionalTorchTensorToDGLArray(csr->data);
+  return aten::CSRMatrix(
+      csr->num_rows, csr->num_cols, indptr, indices, data, csr->sorted);
+}
+
+std::shared_ptr<COO> CSRToCOO(const std::shared_ptr<CSR>& csr) {
+  auto dgl_csr = CSRToOldDGLCSR(csr);
+  auto dgl_coo = aten::CSRToCOO(dgl_csr, csr->data.has_value());
+  return COOFromOldDGLCOO(dgl_coo);
```

Comment:
Have you discussed with other people about how to implement these?

Reply:
These functions convert DGL sparse formats to/from the formats in the sparse library. The purpose is to reuse the implementations in current DGL. What is you concern?

Reply:
I'm fine with reusing the existing implementations, just want to know if this is desired.

Reply:
I see. That is the purpose to link to libdgl.so. We can reduce our development effort and retain the same performance as the message passing interfaces.